### PR TITLE
Remove jsonreader from call

### PR DESCRIPTION
**Pull request by pkuzak**
Resolving technical debt introduced in #316 (_root_reader_ being part of call object).

### CODE REVIEW

**Comment by rfuchs** [comment]
This seems to include the commits from #334 as well. Do you want to close the other pull request?

**Comment by inf265** [comment]
Yes but anyway I don't have the permission I guess ?

**Comment by inf265** [comment]
Ahh I found

File: daemon/call.c [hunk 1]
```diff
@@ -113,37 +114,8 @@ const struct transport_protocol transport_protocols[] = {
...
-};
-static const char *const __opmode_texts[] = {
-	[OP_OFFER] = "offer",
-	[OP_ANSWER] = "answer",
-};
-
-static const char * get_term_reason_text(enum termination_reason t) {
-	return get_enum_array_text(__term_reason_texts, t, "UNKNOWN");
-}
-const char * get_tag_type_text(enum tag_type t) {
```
Comment (rfuchs) [inline daemon/call.c]: Can you also please move the relevant header definitions from `call.h` to `cdr.h`

Comment (inf265) [inline daemon/call.c]: Yes but to be honest I do not very like the parameter handover deep down to every function but I think it's better than before. We'll see if there's a more better way in the future.

**Comment by inf265** [comment]
Yes but to be honest I do not very like the parameter handover deep down to every function but I think it's better than before. We'll see if there's a more better way in the future.

File: daemon/call.c (hunk 1 above)
Comment (inf265) [inline daemon/call.c]: I'll have a look tmorrow and sure I'll move :-D

**Comment by rfuchs** [comment]
Yes I agree, that's why I thought using a global TLS variable would avoid the extra parameter everywhere. At global level, you would declare `static __thread JsonReader *root_reader`, then set this variable at the beginning of the restore process and clear it at the end. Because it's TLS, it's safe to use in all participating functions.

Another option would be to use a helper struct that contains the `call` struct pointer and the `JsonReader` pointer and pass that struct to every function instead of the `call` pointer. You'd just replace `struct call *c` with `struct foobar_helper *x` everywhere and use `x->c` instead of `c` to access the `call`.

Either way, it's up to you. Just some ideas. I'm OK with merging it as it is.

**Comment by inf265** [comment]
Ok if it's ok we leave it as it is. And thanks for explaining the other options.

File: daemon/call.c (hunk 1 above)
Comment (inf265) [inline daemon/call.c]: moved

**Pull request closed without merging**